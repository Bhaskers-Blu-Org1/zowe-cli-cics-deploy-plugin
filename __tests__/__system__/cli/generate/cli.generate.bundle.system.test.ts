/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
* Copyright IBM Corp, 2019
*
*/

import * as fse from "fs-extra";
import * as path from "path";
import {runCliScript} from "../../../__src__/TestUtils";
import {ITestEnvironment} from "../../../__src__/environment/doc/response/ITestEnvironment";
import {TestEnvironment} from "../../../__src__/environment/TestEnvironment";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment;
let TEST_APPS_DIR: string;
let SIMPLE_TEST_APP: string;

describe("cics-deploy generate bundle", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "generate_bundle_command",
            installPlugin: true,
            tempProfileTypes: ["zosmf"]
        });
        TEST_APPS_DIR = path.join(TEST_ENVIRONMENT.workingDir, "apps");
        SIMPLE_TEST_APP = path.join(TEST_APPS_DIR, "simple-node-app");
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    beforeEach(() => {
        if (fse.existsSync(TEST_APPS_DIR)) {
            fse.removeSync(TEST_APPS_DIR);
        }
        fse.mkdirSync(TEST_APPS_DIR);
        fse.copySync(__dirname + "../../../../__resources__/apps", TEST_APPS_DIR);
    });

    it("should generate bundle metadata in the working directory", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/generate_bundle.sh", TEST_ENVIRONMENT, [SIMPLE_TEST_APP]);

        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expectFileExistsContaining(path.join(SIMPLE_TEST_APP, "META-INF", "cics.xml"),
                                         "name=\"cics-nodejs-invoke\"");

        expectFileExistsContaining(path.join(SIMPLE_TEST_APP, "nodejsapps", "cics-nodejs-invoke.nodejsapp"),
                                        "startscript=\"server.js\"");
    });
    describe.skip("permissions tests", () => {
        it("should error nicely if it can't read the working directory", async () => {
            fse.chmodSync(SIMPLE_TEST_APP, "300");
            const response = await runCliScript(__dirname + "/__scripts__/generate_bundle.sh", TEST_ENVIRONMENT, [SIMPLE_TEST_APP]);

            expect(response.status).toBeGreaterThan(0);
        });
        it("should error nicely if it can't write to the working directory", async () => {
            fse.chmodSync(SIMPLE_TEST_APP, "500");
            const response = await runCliScript(__dirname + "/__scripts__/generate_bundle.sh", TEST_ENVIRONMENT, [SIMPLE_TEST_APP]);

            expect(response.status).toBeGreaterThan(0);
        });

        afterEach(() => {
            fse.chmodSync(SIMPLE_TEST_APP, "700");
        });
    });
});

function expectFileExistsContaining(filePath: string, searchString: string) {
    expect(fse.pathExistsSync(filePath));
    const contents = fse.readFileSync(filePath).toString();
    expect(contents).toMatch(searchString);
}

