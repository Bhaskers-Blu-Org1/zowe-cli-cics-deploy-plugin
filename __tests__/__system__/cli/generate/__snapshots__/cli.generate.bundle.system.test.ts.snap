// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cics-deploy generate bundle package.json variations should mangle name supplied in package.json 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"thisisareallylongnamethatwillneedtobetruncatedinordertofitinabun\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"thisisareallylongnamethatwillnee\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/thisisareallylongnamethatwillnee.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle package.json variations should mangle name supplied in package.json 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"thisisareallylongnamethatwillnee\\" startscript=\\"server.js\\" profile=\\"nodejsapps/thisisareallylongnamethatwillnee.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise NODEJSAPP name according to command line args 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"cics-nodejs-invoke\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"myapp\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/myapp.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise NODEJSAPP name according to command line args 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"myapp\\" startscript=\\"server.js\\" profile=\\"nodejsapps/myapp.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise NODEJSAPP name and bundle ID independently 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"mybundle\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"myapp\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/myapp.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise NODEJSAPP name and bundle ID independently 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"myapp\\" startscript=\\"server.js\\" profile=\\"nodejsapps/myapp.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise bundle ID according to command line args 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"myNodeBundle\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"myNodeBundle\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/myNodeBundle.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise bundle ID according to command line args 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"myNodeBundle\\" startscript=\\"server.js\\" profile=\\"nodejsapps/myNodeBundle.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise bundle version according to command line args 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" bundleMajorVer=\\"2\\" bundleMinorVer=\\"1\\" bundleMicroVer=\\"5\\" id=\\"cics-nodejs-invoke\\"><define name=\\"cics-nodejs-invoke\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/cics-nodejs-invoke.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise bundle version according to command line args 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"cics-nodejs-invoke\\" startscript=\\"server.js\\" profile=\\"nodejsapps/cics-nodejs-invoke.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise port in generated profile accord to args 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"cics-nodejs-invoke\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"cics-nodejs-invoke\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/cics-nodejs-invoke.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise port in generated profile accord to args 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"cics-nodejs-invoke\\" startscript=\\"server.js\\" profile=\\"nodejsapps/cics-nodejs-invoke.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should customise port in generated profile accord to args 3`] = `
"#######################################################################
# Node.js application profile: basic_nodejsapp.profile                #
#                                                                     #
# The Node.js application profile is used to configure how CICS       #
# controls the Node.js runtime, provide Node.js command line options, #
# and to set environment variables for use by the application.        #
#                                                                     #
# Lines starting with # are treated as comments and ignored.          #
# Lines ending with \\\\ are continued on the next line.                 #
#                                                                     #
# See topic \\"Node.js profile validation and properties\\"               #
# in the Knowledge Center:                                            #
# http://www.ibm.com/support/knowledgecenter/SSGMCP_5.5.0             #
#                                                                     #
#######################################################################
#
#                      Symbol Substitution
#                      -------------------
#
# Symbols are replaced with their value when the profile is read by
# CICS. They are useful to avoid duplicating configuration in the
# profile. Symbols start with & and end with ; characters. Environment
# variables defined in the profile may also be used as symbols.
#
# The following symbols are provided:
#   &APPLID;     => Applid of the CICS region.
#   &BUNDLE;     => Name of the BUNDLE resource.
#   &BUNDLEID;   => Bundle ID.
#   &CONFIGROOT; => Directory of the .nodejsapp CICS bundle part.
#   &DATE;       => Date when the NODEJSAPP resource is enabled
#                   (localtime), formatted as Dyymmdd.
#   &NODEJSAPP;  => Name of the NODEJSAPP resource.
#   &TIME;       => Time when the NODEJSAPP resource is enabled
#                   (localtime), formatted as Thhmmss.
#   &USSCONFIG;  => Value of the USSCONFIG SIT parameter.
#   &USSHOME;    => Value of the USSHOME SIT parameter.
#
# Examples:
# LOG_DIR=&WORK_DIR;/&APPLID;/&BUNDLEID;/&NODEJSAPP;
# LOG_SUCCESS=&LOG_DIR;/success.txt
# LOG_FAILURE=&LOG_DIR;/failure.txt
#
#**********************************************************************
#
#                      Required parameters
#                      -------------------
#
# NODE_HOME specifies the location of IBM SDK for Node.js - z/OS.
#
# NODE_HOME=/usr/lpp/IBM/cnj/IBM/node-v6.14.4-os390-s390x
#
#
# WORK_DIR specifies the root directory in which CICS will create log
# files. The default value is /tmp. A value of . means the home
# directory of the user id running the CICS job.
#
# WORK_DIR=.
#
#**********************************************************************
#
#                      Including files in the profile
#                      ------------------------------
#
# %INCLUDE specifies a file to be included in this profile. The file can
# contain common system-wide configuration that can then be maintained
# separate to the application configuration. Symbols can be used when
# specifying the file to be included.
#
# Examples:
# %INCLUDE=/etc/cicsts/prodplex/nodejs/sdk.profile
# %INCLUDE=&USSCONFIG;/nodejs/sdk.profile
# %INCLUDE=&CONFIGROOT;/debug.profile
#
%INCLUDE=&USSCONFIG;/nodejsprofiles/general.profile
#
#**********************************************************************
#
#                      Optional parameters
#                      -------------------
#
# NODEJSAPP_DISABLE_TIMEOUT specifies the time in milliseconds that
# CICS will wait when attempting to disable a NODEJSAPP. If the process
# has not terminated in this time then CICS will send a SIGKILL signal
# to force the process to end. The default value is 10000 (10 seconds).
#
# Example:
# NODEJSAPP_DISABLE_TIMEOUT=30000
#
#**********************************************************************
#
#                      Command line options
#                      --------------------
#
# Options beginning with a hyphen will be included on the command line
# when CICS starts the Node.js runtime. The command line options
# accepted by Node.js are documented at:
# https://nodejs.org/dist/latest-v6.x/docs/api/cli.html
#
# Example:
# --trace-sync-io
#
#**********************************************************************
#
#              Unix System Services Environment Variables
#              ------------------------------------------
#
# TZ specifies the local time zone in POSIX format in which the Node.js
# application will run.
#
# Example:
# TZ=CET-1CEST,M3.5.0,M10.5.0
#
#**********************************************************************
#
#                      Environment variables
#                      ---------------------
#
# Environment variables will be set before CICS starts the Node.js
# runtime. Use environment variables to pass configuration to the
# Node.js application.
#
# Example:
# PORT=9080
#
#**********************************************************************
PORT=12345
"
`;

exports[`cics-deploy generate bundle paramters should customise startscript according to args 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"cics-nodejs-invoke\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"cics-nodejs-invoke\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/cics-nodejs-invoke.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should customise startscript according to args 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"cics-nodejs-invoke\\" startscript=\\"other.js\\" profile=\\"nodejsapps/cics-nodejs-invoke.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should generate a bundle using values supplied when there's no package.json 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"mybundle\\"><define name=\\"myapp\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/myapp.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should generate a bundle using values supplied when there's no package.json 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"myapp\\" startscript=\\"server.js\\" profile=\\"nodejsapps/myapp.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should mangle NODEJSAPP name supplied on command line 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"cics-nodejs-invoke\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"fooXXX\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/fooXXX.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should mangle NODEJSAPP name supplied on command line 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"fooXXX\\" startscript=\\"server.js\\" profile=\\"nodejsapps/fooXXX.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle paramters should mangle bundle ID supplied on command line 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"fooXXX\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"fooXXX\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/fooXXX.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle paramters should mangle bundle ID supplied on command line 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"fooXXX\\" startscript=\\"server.js\\" profile=\\"nodejsapps/fooXXX.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle should generate a bundle using defaults from package.json 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\" id=\\"cics-nodejs-invoke\\" bundleMajorVer=\\"1\\" bundleMinorVer=\\"0\\" bundleMicroVer=\\"0\\"><define name=\\"cics-nodejs-invoke\\" type=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" path=\\"nodejsapps/cics-nodejs-invoke.nodejsapp\\"></define></manifest>
"
`;

exports[`cics-deploy generate bundle should generate a bundle using defaults from package.json 2`] = `
"<nodejsapp xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\\" name=\\"cics-nodejs-invoke\\" startscript=\\"server.js\\" profile=\\"nodejsapps/cics-nodejs-invoke.profile\\" lerunopts=\\"DFHSJNRO\\"></nodejsapp>
"
`;

exports[`cics-deploy generate bundle should generate a bundle using defaults from package.json 3`] = `
"#######################################################################
# Node.js application profile: basic_nodejsapp.profile                #
#                                                                     #
# The Node.js application profile is used to configure how CICS       #
# controls the Node.js runtime, provide Node.js command line options, #
# and to set environment variables for use by the application.        #
#                                                                     #
# Lines starting with # are treated as comments and ignored.          #
# Lines ending with \\\\ are continued on the next line.                 #
#                                                                     #
# See topic \\"Node.js profile validation and properties\\"               #
# in the Knowledge Center:                                            #
# http://www.ibm.com/support/knowledgecenter/SSGMCP_5.5.0             #
#                                                                     #
#######################################################################
#
#                      Symbol Substitution
#                      -------------------
#
# Symbols are replaced with their value when the profile is read by
# CICS. They are useful to avoid duplicating configuration in the
# profile. Symbols start with & and end with ; characters. Environment
# variables defined in the profile may also be used as symbols.
#
# The following symbols are provided:
#   &APPLID;     => Applid of the CICS region.
#   &BUNDLE;     => Name of the BUNDLE resource.
#   &BUNDLEID;   => Bundle ID.
#   &CONFIGROOT; => Directory of the .nodejsapp CICS bundle part.
#   &DATE;       => Date when the NODEJSAPP resource is enabled
#                   (localtime), formatted as Dyymmdd.
#   &NODEJSAPP;  => Name of the NODEJSAPP resource.
#   &TIME;       => Time when the NODEJSAPP resource is enabled
#                   (localtime), formatted as Thhmmss.
#   &USSCONFIG;  => Value of the USSCONFIG SIT parameter.
#   &USSHOME;    => Value of the USSHOME SIT parameter.
#
# Examples:
# LOG_DIR=&WORK_DIR;/&APPLID;/&BUNDLEID;/&NODEJSAPP;
# LOG_SUCCESS=&LOG_DIR;/success.txt
# LOG_FAILURE=&LOG_DIR;/failure.txt
#
#**********************************************************************
#
#                      Required parameters
#                      -------------------
#
# NODE_HOME specifies the location of IBM SDK for Node.js - z/OS.
#
# NODE_HOME=/usr/lpp/IBM/cnj/IBM/node-v6.14.4-os390-s390x
#
#
# WORK_DIR specifies the root directory in which CICS will create log
# files. The default value is /tmp. A value of . means the home
# directory of the user id running the CICS job.
#
# WORK_DIR=.
#
#**********************************************************************
#
#                      Including files in the profile
#                      ------------------------------
#
# %INCLUDE specifies a file to be included in this profile. The file can
# contain common system-wide configuration that can then be maintained
# separate to the application configuration. Symbols can be used when
# specifying the file to be included.
#
# Examples:
# %INCLUDE=/etc/cicsts/prodplex/nodejs/sdk.profile
# %INCLUDE=&USSCONFIG;/nodejs/sdk.profile
# %INCLUDE=&CONFIGROOT;/debug.profile
#
%INCLUDE=&USSCONFIG;/nodejsprofiles/general.profile
#
#**********************************************************************
#
#                      Optional parameters
#                      -------------------
#
# NODEJSAPP_DISABLE_TIMEOUT specifies the time in milliseconds that
# CICS will wait when attempting to disable a NODEJSAPP. If the process
# has not terminated in this time then CICS will send a SIGKILL signal
# to force the process to end. The default value is 10000 (10 seconds).
#
# Example:
# NODEJSAPP_DISABLE_TIMEOUT=30000
#
#**********************************************************************
#
#                      Command line options
#                      --------------------
#
# Options beginning with a hyphen will be included on the command line
# when CICS starts the Node.js runtime. The command line options
# accepted by Node.js are documented at:
# https://nodejs.org/dist/latest-v6.x/docs/api/cli.html
#
# Example:
# --trace-sync-io
#
#**********************************************************************
#
#              Unix System Services Environment Variables
#              ------------------------------------------
#
# TZ specifies the local time zone in POSIX format in which the Node.js
# application will run.
#
# Example:
# TZ=CET-1CEST,M3.5.0,M10.5.0
#
#**********************************************************************
#
#                      Environment variables
#                      ---------------------
#
# Environment variables will be set before CICS starts the Node.js
# runtime. Use environment variables to pass configuration to the
# Node.js application.
#
# Example:
# PORT=9080
#
#**********************************************************************
"
`;

exports[`cics-deploy generate bundle should generate an empty bundle if no package.json present 1`] = `
"<manifest xmlns=\\"http://www.ibm.com/xmlns/prod/cics/bundle\\" bundleVersion=\\"1\\" bundleRelease=\\"0\\"></manifest>
"
`;
